lib/logflare/backends/backend.ex

```diff
@@ -42,4 +32,30 @@ defmodule Logflare.Backends.Backend do
 
   @spec child_spec(Source.t(), Backend.t()) :: map()
   defdelegate child_spec(source, backend), to: Adaptor
+
+  # secrets redacting for json encoding
+  defimpl Jason.Encoder, for: __MODULE__ do
+    def encode(value, opts) do
+      type = value.type
+
+      values =
+        Map.take(value, [
+          :name,
+          :token,
+          :description,
+          :type,
+          :id,
+          :config
+        ])
+        |> Map.update(:config, %{}, fn
+          config when type == :postgres ->
+            Map.put(config, :url, "redacted")
```

Comment:
Hmmm, should we store the individual fields instead of the URL?

Will be hard to debug if the whole URL is redacted.

With Realtime and Supavisor we store the fields individually. Could look at other service for prior art too.

Reply:
We could parse URL and the redact only fields that should be secret.

Reply:
will need to keep url in for backwards compat and more flexibility (e.g. adding in query params), but can add the separate config fields as well.

will do the password redaction in string for now, and defer the extra fields to backlog

Reply:
redaction now only removes in-url passwords.